pkg/controller/direct/kms/autokeyconfig_controller.go

```diff
@@ -0,0 +1,273 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package kms
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+
+	krm "github.com/GoogleCloudPlatform/k8s-config-connector/apis/kms/v1alpha1"
+	refs "github.com/GoogleCloudPlatform/k8s-config-connector/apis/refs/v1beta1"
+	"github.com/GoogleCloudPlatform/k8s-config-connector/pkg/config"
+	"github.com/GoogleCloudPlatform/k8s-config-connector/pkg/controller/direct"
+	"github.com/GoogleCloudPlatform/k8s-config-connector/pkg/controller/direct/directbase"
+
+	"github.com/GoogleCloudPlatform/k8s-config-connector/pkg/controller/direct/registry"
+
+	gcp "cloud.google.com/go/kms/apiv1"
+
+	kmspb "cloud.google.com/go/kms/apiv1/kmspb"
+	"google.golang.org/api/option"
+	"google.golang.org/protobuf/types/known/fieldmaskpb"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	ctrlName      = "kms-controller"
+	serviceDomain = "//cloudkms.googleapis.com"
```

Comment:
We made some big improvements in the controller template](https://github.com/GoogleCloudPlatform/k8s-config-connector/blob/master/docs/develop-resources/guides/4-add-controller.md#4-add-the-direct-controller and the `autokeyconfig_externalreference.go` file is no longer needed, so does the serviceDomain. 

Could you update this doc to pick up those changes (either regenerate [the types](https://github.com/GoogleCloudPlatform/k8s-config-connector/blob/master/docs/develop-resources/guides/2-define-apis.md#22-generate-the-config-connector-types-repeat-run-safe) and [controller](https://github.com/GoogleCloudPlatform/k8s-config-connector/blob/master/docs/develop-resources/guides/4-add-controller.md#4-add-the-direct-controller) ) or follow what this [PR](https://github.com/GoogleCloudPlatform/k8s-config-connector/pull/2895/files) does.